### ISSUE TITLE
Any plans to support binlog streaming?

### ISSUE THREAD

**Issue by NxSoftware** [mysql-net/MySqlConnector#687]
Title says it all really, are there any plans to add support for the [Replication Protocol](https://dev.mysql.com/doc/internals/en/replication-protocol.html) so this library could be used in monitoring database changes?

**Comment by bgrainger** [comment]
There are no current plans to add it.

Do you have any idea of what the API would look like? (Is there an example in any other library?)

**Comment by NxSoftware** [comment]
Haven’t given it much thought.

There appears to be no other MySQL library in C# that provides this functionality, the closest analog I have found is in Java: https://github.com/shyiko/mysql-binlog-connector-java/blob/master/README.md

Though I’m not that much of a fan of the way the API works there (iterating and checking the Event Type). I’d probably go with requiring the consumer to register implementations of a generic EventHandler abstract class or interface.

I have started to cobble together some code that parses the initial event but it’s slow going as I’m not familiar with this library nor the MySQL internals 😅

https://github.com/mysql-net/MySqlConnector/compare/master...NxSoftware:master
